src/Elasticsearch.Net/Extensions/UtilExtensions.cs

```diff
@@ -27,6 +27,16 @@ internal static string Utf8String(this MemoryStream ms)
 
 			return Encoding.UTF8.GetString(buffer.Array, buffer.Offset, buffer.Count);
 		}
+		/// <summary> Attempts to return to underlying bytes buffer (no-copy) if possible </summary>
+		internal static byte[] ToArrayOrBuffer(this MemoryStream ms)
+		{
+			if (ms is null) return null;
+
+			if (!ms.TryGetBuffer(out var buffer) || buffer.Array is null)
+				return ms.ToArray();
+
+			return buffer.Array;

```

Comment:
Is it safe to call `TryGetBuffer` on a `RecyclableMemoryStream`? It looks like this method is not overloaded.

Reply:
It's safe in the sense that it works but see my comment on the PR, updating to a later version that actually implements it yields some problems under heavy usage. 

I have a feeling it sometimes writes at an offset which trygetbuffer does not take into account, not been able to trap it yesterday though